source/unifiedbuffer.ts

```diff
@@ -0,0 +1,142 @@
+
+/* spellchecker: disable */
+
+import { Buffer } from './buffer';
+import { Context } from './context';
+import { Initializable } from './initializable';
+
+/* spellchecker: enable */
+
+
+export class UnifiedBuffer extends Initializable {
+
+    protected _cpuBuffer: ArrayBuffer;
+    protected _gpuBuffer: Buffer;
+    protected _updates = new Array<Update>();
+
+    usage: GLenum;
+
+    constructor(context: Context, sizeInBytes: number, usage: GLenum, identifier?: string) {
+        super();
+
+        this._cpuBuffer = new ArrayBuffer(sizeInBytes);
+        this._gpuBuffer = new Buffer(context, identifier);
+        this.usage = usage;
+    }
+
+    protected _addUpdate(update: Update): void {
```

Comment:
Our function names, even for protected functions, do not start with an underscore.
The function name should be `addUpdate`.
Example: [Controller::cancel](https://github.com/cginternals/webgl-operate/blob/master/source/controller.ts#L247)